### PR TITLE
WIP: feat: add js minification (uglify)

### PR DESCRIPTION
**Pull request by makeros**
With minification (and sourcemaps removed) it's 544B vs 4.98kB.
Right now it's not optional but if it makes sense to minify it always then sourcemaps can be removed.

### CODE REVIEW

**Comment by tisba** [comment]
Thank you for your contribution 💕 

Currently source maps are not used yet, so it might make more sense for now to make source maps optional (to save some bytes). That being said, without source map support, an minified test definition would save us a bit of size, but error traces will become next to impossible to decipher :(

Out of curiosity: Are you trying to address a specific problem? Smaller is better, I know, but since the upload using our CLI will be gzipped anyway, so I'm not sure it makes that much of a difference from a user's perspective.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ node_modules/
 # tmp files
 cookie.jar
 tmp/
+dist/
```
Comment (tisba) [inline .gitignore]: Currently we do recommend to checkin the generated test case for easier accountability.

**Review by tisba — CHANGES_REQUESTED** [review]

**Comment by makeros** [comment]
Hi.
I just wanted to save some some bytes ;) But still, gzipped it will be 344B vs 2.28kB.
I will try to enable sourcemaps (even when it's minified).

Another question, did you thought about transform it to a cli tool so that we can install it from npm?

**Comment by makeros** [comment]
Hi @tisba . Sourcemaps and minifying is right now control by arguments.
I've also changed a little bit the gulpfile 😶

**Comment by tisba** [comment]
Hey @makeros!

I'll take a closer look at your changeset ASAP.

Please restore the `dist/` content (`.keep`, etc).

The idea regarding npm is to have a very minimal CLI (or maybe our forge CLI does this) to scaffold the structure and a npm package that defines the gulp pipelines etc. This way we can have a simpler path for updating the tooling.

**Pull request closed without merging**